src/sys/windows/tcp.rs

```diff
@@ -252,6 +252,44 @@ impl TcpStream {
         }
     }
 
+    pub fn peek(&self, buf: &mut [u8]) -> io::Result<usize> {
+        let mut me = self.inner();
+
+        match me.read {
```

Comment:
It looks like this block is copied verbatim from `readv`? Would you mind extracting this to a `before_read` fn or something like that?

Reply:
Done.